src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.ts

```diff
@@ -0,0 +1,164 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+
+import * as _ from 'lodash';
+import { forkJoin as observableForkJoin, of } from 'rxjs';
+import { mergeMap } from 'rxjs/internal/operators';
+
+import { ConfigurationService } from '../../../../shared/api/configuration.service';
+import { NotificationType } from '../../../../shared/enum/notification-type.enum';
+import { CdFormGroup } from '../../../../shared/forms/cd-form-group';
+import { NotificationService } from '../../../../shared/services/notification.service';
+import { OsdRecvSpeedProfiles } from './osd-recv-speed-modal.profiles';
+
+@Component({
+  selector: 'cd-osd-recv-speed-modal',
+  templateUrl: './osd-recv-speed-modal.component.html',
+  styleUrls: ['./osd-recv-speed-modal.component.scss']
+})
+export class OsdRecvSpeedModalComponent implements OnInit {
+  osdRecvSpeedForm: CdFormGroup;
+  profiles = OsdRecvSpeedProfiles.KNOWN_PROFILES;
+  profileAttrs = [
+    'osd_max_backfills',
+    'osd_recovery_max_active',
+    'osd_recovery_max_single_start',
+    'osd_recovery_sleep'
+  ];
+
+  constructor(
+    public bsModalRef: BsModalRef,
+    private configService: ConfigurationService,
+    private notificationService: NotificationService
+  ) {
+    this.osdRecvSpeedForm = new CdFormGroup({
+      profile: new FormControl(null, { validators: [Validators.required] }),
+      customizeProfile: new FormControl(false)
+    });
+    this.profileAttrs.forEach((attr) => {
+      this.osdRecvSpeedForm.addControl(
+        attr,
+        new FormControl(null, { validators: [Validators.required] })
+      );
+    });
+  }
+
+  ngOnInit() {
+    this.getStoredProfile((profile) => {
+      this.setProfile(profile);
+    });
+  }
+
+  setProfile(profile: any) {
+    const customProfile = _.find(this.profiles, (p) => {
+      return p.name === 'custom';
+    });
+
+    if (profile.name === 'custom') {
+      if (!customProfile) {
+        this.profiles.push(profile);
+      }
+    } else {
+      if (customProfile) {
+        this.profiles.splice(this.profiles.indexOf(customProfile), 1);
+      }
+    }
+
+    this.osdRecvSpeedForm.controls.profile.setValue(profile.name);
+    Object.entries(profile.values).forEach(([name, value]) => {
+      this.osdRecvSpeedForm.controls[name].setValue(value);
+    });
+  }
+
+  onCustomizeProfileChange() {
+    if (this.osdRecvSpeedForm.getValue('customizeProfile')) {
+      const values = {};
+      this.profileAttrs.forEach((attr) => {
+        values[attr] = this.osdRecvSpeedForm.getValue(attr);
+      });
+      const customProfile = {
+        name: 'custom',
+        text: 'Custom',
```

Comment:
Urgh... good question. I don't think this would be a good idea as I'm using it here: https://github.com/ceph/ceph/pull/24968/files#diff-e6f9474d1d71f5d57be376f98e8c85caR126
Do you have an idea how we should handle these things?

Reply:
Any string that is show in a page should be translated.
```suggestion
        text: this.i18n('Custom'),
```